src/components/Movies.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import Movie from './Movie';
+import FavMovie from './FavMovie';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+
+
+
+
+
+class Movies extends React.Component{
+  constructor(){
+    super();
+    this.state={
+      favList:[],
+      displayFav:false,
+
+
+    }
+    this.receiveFavClick = this.receiveFavClick.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.receiveDeleteClick = this.receiveDeleteClick.bind(this);
+  }
+
+  componentDidMount(){
+    const storageString = window.localStorage.getItem('favList');
+    const localFavList = !storageString ? []: JSON.parse(storageString)
+    this.setState({
+      favList: localFavList,
+    })
+  }
+
+  receiveFavClick(id){
+
+    if(this.state.favList.filter(movie => movie.imdbID==id).length == 0){
+        const newMovie=this.props.movies.filter(movie => movie.imdbID == id);
+        this.setState({
+          favList: this.state.favList.concat(newMovie),
+        },()=>localStorage.setItem("favList", JSON.stringify(this.state.favList))
+      )
+    } else {
+        this.setState({
+          favList: this.state.favList.filter(movie => movie.imdbID !== id),
+      }, ()=>localStorage.setItem('favList',JSON.stringify(this.state.favList))
+    )}
+  }
+
+  receiveDeleteClick(id){
+    this.setState({
+      favList: this.state.favList.filter(movie => movie.imdbID !== id),
+
+  }, ()=>localStorage.setItem('favList',JSON.stringify(this.state.favList))
+);
+  }
+
+
+  handleClick(event){
+    this.setState({
+      displayFav:!this.state.displayFav
+    })
+  }
+
+  render(){
+    const storageString=localStorage.getItem('favList')
```

Comment:
It would better to use `favList` from state rather than getting it from localStorage. `favList` in state should have same items.